packages/netlify-cms-core/src/actions/mediaLibrary.js

```diff
@@ -186,14 +189,39 @@ export function persistMedia(file, opts = {}) {
       const id = await getBlobSHA(file);
       const assetProxy = await createAssetProxy(fileName, file, false, privateUpload);
       dispatch(addAsset(assetProxy));
+
+      const entry = state.entryDraft.get('entry');
+      const useWorkflow = state.config.getIn(['publish_mode']) === EDITORIAL_WORKFLOW;
+      const draft = entry && !entry.isEmpty() && useWorkflow;
```

Comment:
I'm not sure this is the best way to check that we are editing an entry

Reply:
I would expect the entry in `state.entryDraft` to always be an open entry, but maybe there are gaps - did you experience this not being the case?

Reply:
I'm note sure what "open entry" means.
What I have experienced is that when you exit the editor the draft is discarded here: https://github.com/netlify/netlify-cms/blob/7554134f999970e8f8198f38200c47ff5880c0a4/packages/netlify-cms-core/src/reducers/entryDraft.js#L72